### PR TITLE
fix: Clean up key bindings when leaving rundown view.

### PR DESCRIPTION
**Pull request by KvelaGorrrrnio**
Fixes issue where key bindings were triggered for a rundown previously, but no longer, shown in the rundown view.

### CODE REVIEW

**Review by RasmusAlbrektsen — APPROVED** [review]

**Review by LindvedKrvang — APPROVED** [review]

File: src/app/rundown-view/services/action-trigger-producer-key-binding.service.ts [hunk 1]
```diff
@@ -35,7 +35,7 @@ export class ActionTriggerProducerKeyBindingService implements KeyBindingService
   private rundown?: Rundown
   private systemKeyBindings: KeyBinding[] = []
   private actionTriggerKeyBindings: KeyBinding[] = []
-  private readonly keyBindingsSubject: Subject<KeyBinding[]>
+  private keyBindingsSubject: Subject<KeyBinding[]>
```
Comment (MadsThy) [inline src/app/rundown-view/services/action-trigger-producer-key-binding.service.ts]: You are assigning this to be a behaviorSubject further down :D I propose this change instead:

```suggestion
  private keyBindingsSubject = new BehaviorSubject<KeyBinding[]>(this.systemKeyBindings)
```

And then you can delete line 59

File: src/app/rundown-view/services/action-trigger-producer-key-binding.service.ts [hunk 2]
```diff
@@ -277,13 +280,18 @@ export class ActionTriggerProducerKeyBindingService implements KeyBindingService
   }
 
   public destroy(): void {
+    this.systemKeyBindings = []
     this.keyBindingsSubject.complete()
+    this.keyBindingsSubject = new BehaviorSubject(this.systemKeyBindings)
```
Comment (MadsThy) [inline src/app/rundown-view/services/action-trigger-producer-key-binding.service.ts:284]:
```suggestion
    this.keyBindingsSubject.next(this.systemKeyBindings)
```

I don't know what purpose it serves to call complete to then directly update it afterwards 😄 Will this change do?

**Review by MadsThy — APPROVED** [review]

File: src/app/rundown-view/services/action-trigger-producer-key-binding.service.ts (hunk 2 above)
Comment (KvelaGorrrrnio) [reply src/app/rundown-view/services/action-trigger-producer-key-binding.service.ts:284]: The purpose is that when we call destroy, we should ensure that updates are no longer propagated to old consumers.

File: src/app/rundown-view/services/action-trigger-producer-key-binding.service.ts (hunk 1 above)
Comment (KvelaGorrrrnio) [reply src/app/rundown-view/services/action-trigger-producer-key-binding.service.ts]: It is updated to suggestion :)

**Pull request merged**